config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/IdentityProviderEntityBuilder.java

```diff
@@ -62,41 +67,61 @@ private Entity buildIdentityProviderEntity(String name, IdentityProvider identit
         return new Entity("ID_PROVIDER_CONFIG", name, id, identityProviderElement);
     }
 
+    private Element buildFedIdProviderDetails(FederatedIdentityProviderDetail identityProviderDetail) {
+        final Element extensionElement = document.createElement(EXTENSION);
+        final Element federatedIdProviderDetailElem = document.createElement(FEDERATED_ID_PROV_DETAIL);
+        extensionElement.appendChild(federatedIdProviderDetailElem);
+        final Element certReferencesElem = DocumentTools.createElementWithAttribute(
+                document,
+                CERTIFICATE_REFERENCES,
+                ATTRIBUTE_RESOURCE_URI,
+                TRUSTED_CERT_URI);
+        federatedIdProviderDetailElem.appendChild(certReferencesElem);
+        final List<String> certReferences = identityProviderDetail.getCertificateReferences();
+        if (certReferences == null || certReferences.isEmpty()) {
```

Comment:
you can use CollectionUtils.isNotEmpty that already checks for null and makes your if simpler :)